### PR TITLE
Fixing lint errors in files touched by PR #338

### PR DESCRIPTION
**Pull request by thk123**
Corrected lint errors in some files as they were touched by #338.

There are some outstanding errors that I wasn't sure how to fix:

- an anonomous namespace in java_entry_point.cpp
- The formatting of the CBMC help message

There are also some bugs with the linter incorrectly reporting errors.

### CODE REVIEW

**Comment by thk123** [comment]
In [`void cbmc_parse_optionst::help()`](https://github.com/diffblue/cbmc/pull/343/files#diff-41db988013f70caad45ce489749c7df6R1047) multiple lines extend over 80chars but I think it will make the code strictly less readable to correct this. Perhaps we need a way to add exceptions to rules for specific functions so we can still fail on errors.

**Comment by thk123** [comment]
Bugs in the linter:

> If/else bodies with multiple statements require braces  [readability/braces] [4]

[Caused by](https://github.com/diffblue/cbmc/pull/343/files#diff-41db988013f70caad45ce489749c7df6R471):

    if(cmdline.isset("module") ||
       cmdline.isset("gen-interface"))
  
    {
      error() << "This version of CBMC has no support for "
                 " hardware modules. Please use hw-cbmc." << eom;
      return 1; // should contemplate EX_USAGE from sysexits.h
    }

Which does have braces...

> Using C-style cast.  Use static_cast<int>(...) instead  [readability/casting] [4]

[Caused by:](https://github.com/diffblue/cbmc/pull/343/files#diff-41db988013f70caad45ce489749c7df6R990)

    catch(int)

The int isn't being used so naming it will produced an unused variable warning. 

>  Remove spaces around +  [whitespace/operators] [4]

[Caused by:](https://github.com/diffblue/cbmc/pull/343/files#diff-41db988013f70caad45ce489749c7df6R680)

    ++it;

I reported this elsewhere, it trips up in for loops as well.

**Comment by martin-cs** [comment]
On Thu, 2016-12-08 at 05:31 -0800, thk123 wrote:
> In [`void cbmc_parse_optionst::help()`](https://github.com/diffblue/cbmc/pull/343/files#diff-41db988013f70caad45ce489749c7df6R1047) multiple lines extend over 80chars but I think it will make the code strictly less readable to correct this. Perhaps we need a way to add exceptions to rules for specific functions so we can still fail on errors. 

(This is exactly the kind of thing I was referring to when I suggested
that we automate the coding standard and role out a first version rather
than debating it too much.  This discussion is progress and IMHO more
efficient than having this discussion up front.)

Yes; this sounds like a good idea.

Cheers,
 - Martin

File: src/java_bytecode/java_entry_point.cpp [hunk 1]
```diff
@@ -46,32 +47,33 @@ Function: create_initialize
 
 \*******************************************************************/
 
-namespace {
```
Comment (thk123) [inline src/java_bytecode/java_entry_point.cpp]: The linter says no namespaces, I'm not sure if this includes anon namespaces, and if so what should be done about this function. Probably should end up in a class per this refactor suggestion: https://github.com/diffblue/cbmc/pull/338#issuecomment-265497399

Comment (peterschrammel) [inline src/java_bytecode/java_entry_point.cpp]: yes, this should be in a class. you can make it static as an intermediate step before someone refactors it

**Comment by thk123** [comment]
Looks like the cast can be fixed on this line: https://github.com/diffblue/cbmc/blob/master/scripts/cpplint.py#L5273

I didn't spot before, but the braces is a legitimate error, there is an unused empty line after the if which is confusing.

File: src/ansi-c/ansi_c_entry_point.cpp [hunk 2]
```diff
@@ -194,7 +196,8 @@ bool ansi_c_entry_point(
     forall_symbol_base_map(it, symbol_table.symbol_base_map, config.main)
     {
       // look it up
-      symbol_tablet::symbolst::const_iterator s_it=symbol_table.symbols.find(it->second);
+      symbol_tablet::symbolst::const_iterator s_it=symbol_table.symbols.find(
+        it->second);
```
Comment (tautschnig) [inline src/ansi-c/ansi_c_entry_point.cpp]: We mostly seem to break the line after the assignment.

File: src/ansi-c/ansi_c_entry_point.cpp [hunk 3]
```diff
@@ -109,15 +109,17 @@ void record_function_outputs(
   code_blockt &init_code,
   symbol_tablet &symbol_table)
 {
-  //const code_typet::parameterst &parameters=
-  //  to_code_type(function.type).parameters();
+#if 0
+  const code_typet::parameterst &parameters=
+  to_code_type(function.type).parameters();
+#endif
```
Comment (tautschnig) [inline src/ansi-c/ansi_c_entry_point.cpp]: I don't understand this one - why isn't that simply removed?

File: src/java_bytecode/java_entry_point.cpp [hunk 4]
```diff
@@ -30,7 +29,9 @@ Author: Daniel Kroening, kroening@kroening.com
 #include "java_entry_point.h"
 #include "java_object_factory.h"
 
-//#include "zero_initializer.h"
+#if 0
+#include "zero_initializer.h"
+#endif
```
Comment (tautschnig) [inline src/java_bytecode/java_entry_point.cpp]: Just remove this, there is no value in keeping it: 1) a header will be included when necessary; 2) we have git.

**Comment by thk123** [comment]
@tautschnig Thanks for the careful feedback. I have addressed those 3 changes (my logic with the commenting was since was just fixing the linting errors and I didn't know why they'd be commented out but left in was safer to just leave them in, but I've removed them now).

File: src/cbmc/cbmc_parse_options.cpp [hunk 5]
```diff
@@ -239,23 +242,29 @@ void cbmc_parse_optionst::get_command_line_options(optionst &options)
...
+  options.set_option(
+    "partial-loops",
+    cmdline.isset("partial-loops"));
 
   if(options.get_bool_option("partial-loops") &&
      options.get_bool_option("unwinding-assertions"))
   {
-    error() << "--partial-loops and --unwinding-assertions must not be given together" << eom;
+    error() << "--partial-loops and --unwinding-assertions ";
+    error() << "must not be given together" << eom;
```
Comment (kroening) [inline src/cbmc/cbmc_parse_options.cpp]: No need to report error()

File: src/cbmc/cbmc_parse_options.cpp [hunk 6]
```diff
@@ -841,8 +859,8 @@ bool cbmc_parse_optionst::process_goto_program(
     remove_asm(symbol_table, goto_functions);
 
     // add the library
-    status() << "Adding CPROVER library ("
-             << config.ansi_c.arch << ")" << eom;
+    status() << "Adding CPROVER library (";
+    status() << config.ansi_c.arch << ")" << eom;
```
Comment (kroening) [inline src/cbmc/cbmc_parse_options.cpp]: The original code seems ok!

File: scripts/cpplint.py [hunk 7]
```diff
@@ -5270,7 +5270,7 @@ def CheckCStyleCast(filename, clean_lines, linenum, cast_type, pattern, error):
 
   # Exclude lines with keywords that tend to look like casts
   context = line[0:match.start(1) - 1]
-  if Match(r'.*\b(?:sizeof|alignof|alignas|[_A-Z][_A-Z0-9]*)\s*$', context):
+  if Match(r'.*\b(?:sizeof|alignof|alignas|catch|[_A-Z][_A-Z0-9]*)\s*$', context):
```
Comment (peterschrammel) [inline scripts/cpplint.py]: I'll add this to https://github.com/diffblue/cbmc/pull/342; this PR should only touch source files

Comment (thk123) [inline scripts/cpplint.py]: Done.

File: src/cbmc/cbmc_parse_options.cpp [hunk 8]
```diff
@@ -310,7 +321,8 @@ void cbmc_parse_optionst::get_command_line_options(optionst &options)
 
   if(cmdline.isset("smt2"))
   {
-    options.set_option("smt1", false);// If both are given, smt2 takes precedence
+    // If both are given, smt2 takes precedence
+    options.set_option("smt1", false);
     options.set_option("smt2", true);
     version_set = true;
```
Comment (thk123) [inline src/cbmc/cbmc_parse_options.cpp]: spaces around =

**Comment by thk123** [comment]
Don't merge this just yet - I am going to try running on an updated version of the linter (as in #342) and 

a) check all the false positives are gone
b) add exceptions via a comment in the source to remaining errors

**Pull request merged**